src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -30,17 +36,28 @@ import { fmt_to_fragments } from 'utils.jsx';
 import { NeedsShutdownAlert } from '../common/needsShutdown.jsx';
 import { domainReplaceSpice } from '../../libvirtApi/domain.js';
 
+import './vmReplaceSpiceDialog.scss';
+
 const _ = cockpit.gettext;
 
-export const ReplaceSpiceDialog = ({ vm }) => {
+export const ReplaceSpiceDialog = ({ vm, vms }) => {
+    const spiceVMs = vms?.filter(vm => vm.inactiveXML?.hasSpice);
+    // sort spicVMs by name, put the current VM first
+    spiceVMs?.sort((a, b) => (a.name === vm.name) ? -1 : (b.name === vm.name) ? 1 : a.name.localeCompare(b.name));
+    const isMultiple = spiceVMs?.length > 1;
+
     const Dialogs = useDialogs();
     const [error, dialogErrorSet] = useState(null);
     const [inProgress, setInProgress] = useState(false);
+    const [selected, setSelected] = useState(isMultiple ? [spiceVMs[0].id] : []);
 
     function onReplace() {
         setInProgress(true);
 
+        const selectedVMs = spiceVMs?.filter(vm => selected.includes(vm.id)) || [];
+
         return domainReplaceSpice({ connectionName: vm.connectionName, id: vm.id })
```

Comment:
So the primary VM always gets replaced, eventhough there is a checkbox for it in the multi case that the user might uncheck?

Reply:
Indeed, thanks for spotting! I created an integration test for covering this case. The dialog is rather complicated due to the possibility of unchecking the VM which you triggered the action on :cry: